kernel/src/thread/task.rs

```diff
@@ -78,7 +82,12 @@ pub fn create_new_user_task(user_space: Arc<UserSpace>, thread_ref: Arc<Thread>)
             if current_thread.is_exited() {
                 break;
             }
-            handle_pending_signal(user_ctx, &ctx, syscall_number).unwrap();
+            if let Err(e) = handle_pending_signal(user_ctx, &ctx, syscall_number) {
+                debug!("Failed to handle pending signal: {:?}", e);
+                let term_status = TermStatus::Exited(u8::MAX);
```

Comment:
Could you add a comment to explain why the exit code should be `u8::MAX`, which is not apparent to me.

Reply:
I use `u8::MAX` here because if a program exit with error, it usually will return -1. 
This error code is commonly used for generic errors

Do you have any suggestion?

Reply:
I think you should use `TermStatus::Killed(SIGKILL)`. This is not a normal exit. It is more like the kernel kills the thread. And I don't know if you should simply call `exit_group` to kill the whole process. Seems reasonable to me.

And internally, `handle_pending_signal` also has exit group logic, because it may handle SIGKILL or other terminating signals anyway. So I think you probably want to move the exit-the-thread-if-unable-to-handle-user-signal logic into `handle_pending_signal`.

Reply:
Thank you for the guidance!

> And I don't know if you should simply call exit_group to kill the whole process.
I tried to use `current_thread.exit()` here, but it falls into an infinity loop in `init_thread()` then.